modules/aws/vpc.go

```diff
@@ -143,6 +143,43 @@ func GetSubnetsForVpcE(t *testing.T, vpcID string, region string) ([]Subnet, err
 	return subnets, nil
 }
 
+// IsPublicSubnet returns True if the subnet identified by the given id in the provided region is public.
+func IsPublicSubnet(t *testing.T, subnetId string, region string) bool {
+	isPublic, err := IsPublicSubnetE(t, subnetId, region)
+	require.NoError(t, err)
+	return isPublic
+}
+
+// IsPublicSubnetE returns True if the subnet identified by the given id in the provided region is public.
+func IsPublicSubnetE(t *testing.T, subnetId string, region string) (bool, error) {
+	subnetIdFilterName := "association.subnet-id"
+
+	subnetIdFilter := ec2.Filter{
+		Name:   &subnetIdFilterName,
+		Values: []*string{&subnetId},
+	}
+
+	client, err := NewEc2ClientE(t, region)
+	if err != nil {
+		return false, err
+	}
+
+	rts, err := client.DescribeRouteTables(&ec2.DescribeRouteTablesInput{Filters: []*ec2.Filter{&subnetIdFilter}})
+	if err != nil {
+		return false, err
+	}
+
+	for _, rt := range rts.RouteTables {
+		for _, r := range rt.Routes {
+			if strings.HasPrefix(*aws.String(*r.GatewayId), "igw-") {
```

Comment:
Sorry I just realized I gave the wrong function! This should be:

```suggestion
			if strings.HasPrefix(aws.StringValue(r.GatewayId), "igw-") {
```

Reply:
done